torch/fx/experimental/GraphManipulation.py

```diff
@@ -24,3 +27,62 @@ def replace_target_nodes_with(
         else:
             val_map[node] = new_graph.node_copy(node, lambda n : val_map[n])
     fx_module.graph = new_graph
+
+class NodeSize(NamedTuple):
+    output_size: int
+    total_size: int
+
+def get_size_of_all_nodes(fx_module: GraphModule, args: List[torch.Tensor]) -> None:
+    """Given a fx graph module, update each node with its total size (weights + bias + output)
+    and its output_size(output). For a non-module node, the total size is the output size.
+    return total size"""
+    # Mark shape and dtype for each node (node.shape and node.dtype)
+    ShapeProp(fx_module).propagate(*args)
+    # Calculate the total size of the whole fx graph
+    total_size_of_graph = 0.0
+    for node in fx_module.graph.nodes:
+        if node.name == 'output':
+            break
+        node.NodeSize = get_size_of_node(fx_module, node)
+    return
+
+def get_size_of_node(fx_module: GraphModule, node: Node) -> NodeSize:
+    """Given a node with node.dtype and node.shape, return its total size and its output size.
+       total_size = weights + bias + output_size
+    """
+    def product_of_list(size_list: List[int]):
+        product = 1
+        for i in size_list:
+            product *= i
+        return product
+
+    # Total num of elements (weights + bias + output)
+    total_num_of_elems = 0
+    # For a module, conside weights and bias
+    if node.op == 'call_module':
+        submodule_dict = dict(fx_module.named_modules())
+        if node.name in submodule_dict:
+            submodule = submodule_dict[node.name]
+        else:
+            tokens = node.name.split('_')
+            module_name = '_'.join(tokens[:-1])
+            submodule = submodule_dict[module_name]
+        parameters = submodule.named_parameters()
+        # Parameters are named tuples
+        for p in parameters:
```

Comment:
`for name, p in parameters` will unpack the key, value pairs in `named_parameters`

Reply:
This is still outstanding, unpacking in the loop declaration is more idiomatic and `p[1]` below will be replace with `p`